robot/rover/PDS/PDS.ino

```diff
@@ -295,15 +355,126 @@ float readMultiplexer(uint8_t channel) {
     channel = 0; //! \todo not the best implementation, should return a negative value or a false bool
   }
 
-  //! this is done because S0 and S1 are swapped on PORTD. This swaps those bits in the channel variable
-  uint8_t bit0 = (channel >> 0) & 1; // save bit 0
-  uint8_t bit1 = (channel >> 1) & 1; // save bit 1
-  uint8_t mask = bit0 ^ bit1; // if 11 or 00 then mask is 0, else mask is 1
-  mask = (mask << 0) | (mask << 1); // turn mask of 0 into 00, turn mask of 1 into 11
-  channel = channel ^ mask; // 0 XOR 0 preserves the bit, anything XOR 1 flips the bit, so if mask is 11 it flips the lower 2 bits and regardless it preserves everything else
-
-  // set bits two to five on PORTD for mux select line
-  PORTD = (PORTD & B11000011) | (channel << 2);
+  // //! this is done because S0 and S1 are swapped on PORTD. This swaps those bits in the channel variable
+  // uint8_t bit0 = (channel >> 0) & 1; // save bit 0
+  // uint8_t bit1 = (channel >> 1) & 1; // save bit 1
+  // uint8_t mask = bit0 ^ bit1; // if 11 or 00 then mask is 0, else mask is 1
+  // mask = (mask << 0) | (mask << 1); // turn mask of 0 into 00, turn mask of 1 into 11
+  // channel = channel ^ mask; // 0 XOR 0 preserves the bit, anything XOR 1 flips the bit, so if mask is 11 it flips the lower 2 bits and regardless it preserves everything else
+
+  // // set bits two to five on PORTD for mux select line
+  // PORTD = (PORTD & B11000011) | (channel << 2);
+
+  switch (channel) {
+    case 0:
```

Comment:
This whole switch can be converted into four lines if by just converting channel into binry and using a bitwise operation.

Reply:
As you and I discussed, we will be leaving it as is. Keeps the code simpler for anyone looking at it.